### ISSUE TITLE
Winning Shapes Calculator

### ISSUE THREAD

**Issue by geurtjens** [geurtjens/CrozzleRectangleMaker#43]
We will rename the `StrategyCalculator` to `WinningShapesCalculator`

Each routine will be split into getting the winning shape components and secondly creating the winning shape.

So we can utilise the components or individual winning shapes in other ways like 
* seeing if the winning shapes are present in the current calculation.
* using only the winning shapes for various algorithms so as to avoid shapes that don't win, thus speeding up tests.

Basically the main algorithm identifies winning shapes and then merges them so starting with only winning shapes means we can focus on the merge of multiple shapes and knowing we have already found the winning shapes.

Secondly to speed it up we can have a variable which tells if we want the debug information.  By default debug information is not printed but in our tests it can be printed.  This debug info was very useful in composing winning shapes.

**Comment by geurtjens** [comment]
Converted the code from just creating the winning shape to providing the list of shapes that make up the winning shape.

**Comment by geurtjens** [comment]
Implemented.  Have yet to use these winning shapes but they are available.
